Pass monitor name as optional description to $XPLUGRC.

magnusmalm opened the pull request:
This pull requst adds monitor name to $XPLUGRC as "optional description" (fourth argument).

Monitor name is extracted from the connected monitor's EDID data. If no such data is available for whatever reason (old monitor model, or incorrect header/checksum), an empty string is passed instead.

Tests have only been performed with one monitor connected at a time.

magnusmalm committed:
Pass monitor name as optional description to $XPLUGRC.

Extract monitor name from EDID data (if available) and pass it to
$XPLUGRC as fourth argument.

Signed-off-by: Magnus Malm <magnusmalm@gmail.com>

troglobit commented:
Cool, you managed to get some time over to implement this, awesome! :sunglasses::+1: 

I only have a few comments, that I'll leave in the review.  You can just add a few follow-up commits to your branch and this pull-request will pick them up automatically.

troglobit commented on edid-parse.c:
```diff
@@ -0,0 +1,145 @@
+/*
```
This file should be in same coding style as the rest of the xplugd project.  This is less important than the other points (below), but I'm mentioning it anyway if you get the time to call [indent](https://github.com/troglobit/toolbox/blob/master/indent.sh) on it.

troglobit commented on edid-parse.c:
```diff
@@ -0,0 +1,145 @@
+#include <string.h>
+
+#define TRUE 1
+#define FALSE 0
```
I've never really understood the need for this. Sure, it's a practise that some projects maintain, but why not just stick to POSIX style, i.e. return 0 for OK and non-zero on error?

troglobit commented on edid-parse.c:
```diff
@@ -0,0 +1,145 @@
+	if (memcmp(edid, "\x00\xff\xff\xff\xff\xff\xff\x00", 8) == 0)
+		return TRUE;
+	return FALSE;
+}
```
This function for instance, which returns `TRUE` or `FALSE`, could just as easily have been named `is_edid_header()` to indicate it's a predicate function that returns non-zero when true.

troglobit commented on edid-parse.c:
```diff
@@ -0,0 +1,145 @@
+
+MonitorInfo *decode_edid(const uchar *edid)
+{
+	MonitorInfo *info = calloc(1, sizeof(MonitorInfo));
```
Here's the really bad part. Memory is allocated but never freed (as far as I can see).

troglobit commented on randr.c:
```diff
@@ -21,15 +21,51 @@
 
 static char *con_actions[] = { "connected", "disconnected", "unknown" };
 
+int get_monitor_name(char *name, char *monitor_name)
```
Another, semi-serious, issue is this: `char *name` should probably be `const char *name` and the pointer to the resulting string, `monitor_name`, should really have an accompanying extra argument `size_t len` instead of hard-coding 14 in the `strncpy()`.

troglobit commented on randr.c:
```diff
@@ -21,15 +21,51 @@
+		return -1;
+	}
+
+	result = fread (edid, 1, sizeof(edid), fp);
```
Minor nitpick, no space between `fread` an `(` ...

troglobit commented on randr.c:
```diff
@@ -21,15 +21,51 @@
+
+	syslog(LOG_DEBUG, "MODEL: %s\n", info->dsc_product_name);
+	strncpy(monitor_name, info->dsc_product_name, 14);
+
```
I guess this is where we could do `free(info);` right?

troglobit commented on randr.c:
```diff
@@ -21,15 +21,51 @@
+	}
+
+	info = decode_edid(edid);
+	if (info == NULL) {
```
You changed style here, above you used `if (!fp)`, not `if (fp == NULL)` ... consistency is always good. Point of note, I prefer the former: `if (!info)` in this case.

troglobit commented on randr.c:
```diff
@@ -21,15 +21,51 @@
+		return -1;
+	}
+
+	syslog(LOG_DEBUG, "MODEL: %s\n", info->dsc_product_name);
```
In `syslog()` messages you don't need the extra line-feed `\n`.

troglobit commented on randr.c:
```diff
@@ -21,15 +21,51 @@
+
+	result = fread (edid, 1, sizeof(edid), fp);
+	if (result != 128) {
+		syslog(LOG_NOTICE, "No EDID data found");
```
A user with a faulty display, or similar, would see this message very often.  How about changing this to `LOG_INFO`, or maybe even collapse the two messages into one `LOG_DEBUG`?

When debugging the setup, the first couple of times you run xplugd in a new environment, I'm guessing most people enable all debug they can get.

troglobit commented on randr.c:
```diff
@@ -21,15 +21,51 @@
+	result = fread (edid, 1, sizeof(edid), fp);
+	if (result != 128) {
+		syslog(LOG_NOTICE, "No EDID data found");
+		syslog(LOG_DEBUG, "DRM device sysfs path %s\n", path);
```
In `syslog()` messages you don't need the extra line-feed `\n`.

troglobit commented on randr.c:
```diff
@@ -21,15 +21,51 @@
+	MonitorInfo *info;
+
+	snprintf(path, sizeof(path), "/sys/class/drm/card0-%s/edid", name);
+	syslog(LOG_DEBUG, "DRM device sysfs path %s\n", path);
```
In `syslog()` messages you don't need the extra line-feed `\n`.

troglobit commented on randr.c:
```diff
@@ -21,15 +21,51 @@
+	syslog(LOG_DEBUG, "DRM device sysfs path %s\n", path);
+	fp = fopen(path, "rb");
+	if (!fp) {
+		syslog(LOG_ERR, "Failed to find sys path for %s\n", name);
```
In `syslog()` messages you don't need the extra line-feed `\n`.

magnusmalm committed:
Fix coding style on new files and in randr.c.

Also, Consistency re NULL check and remove space between fread and (
in get_monitor_name().

Signed-off-by: Magnus Malm <magnusmalm@gmail.com>

magnusmalm committed:
Be more POSIX re predicate functions and return values.

Signed-off-by: Magnus Malm <magnusmalm@gmail.com>

magnusmalm committed:
Fix memory leak for Monitorinfo in decode_edid().

Signed-off-by: Magnus Malm <magnusmalm@gmail.com>

magnusmalm committed:
Remove hard-coded len to strncpy in get_monitor_name().

Signed-off-by: Magnus Malm <magnusmalm@gmail.com>

magnusmalm committed:
Cleanup syslog messages.

Signed-off-by: Magnus Malm <magnusmalm@gmail.com>

magnusmalm commented:
All excellent comments. Much appreciated! I've addressed all of them (I hope).

I probably should reply to each comment with a ref to corresponding commit but the hour is late and the number and size of the commits are small so I hope you are ok with that. :smile:

troglobit commented:
Ah, hackers night-shift! :rofl: 

This looks so much better, thanks for taking the time to clean it up! :+1: 

I've just got a minor comment, or two, which is mostly for future reference.

troglobit commented on edid-parse.c:
```diff
@@ -26,42 +26,33 @@
 }
 
-static void decode_display_descriptor(const uchar *desc, MonitorInfo *info)
+static void decode_display_descriptor(const uchar *desc, MonitorInfo * info)
```
Here's one of the caveats with GNU indent.  When it doesn't know of a composed type it inserts a gratuitous extra space here.  It's soooooo annoying and the only two ways of getting around it, that I know of is:

1. Add the type to the indent command line, `indent -T MonitorInfo ...`, I've added a few to my own wrapper script, which you may have noticed.
2. Do it like the kernel does, which is what I'm beginning to lean more and more to myself, i.e. don't use `typedef` but instead use `struct MonitorInfo` everywhere, because *that* GNU indent understands and does right by ... *sigh*

troglobit commented on edid-parse.c:
```diff
@@ -94,24 +85,22 @@ static void decode_display_descriptor(const uchar *desc, MonitorInfo *info)
 }
 
-static int decode_descriptors(const uchar *edid, MonitorInfo *info)
+static int decode_descriptors(const uchar *edid, MonitorInfo * info)
```
Same gratuitous extra space here ...

troglobit commented:
Now we'll just have to wait for Travis-CI to fix their current build issues.

troglobit commented:
There, seems they've fixed their massive backlog. 

Merging ... again, great work on this, thanks! :beers:
